pennylane/labs/trotter_error/product_formulas/error.py

```diff
@@ -36,9 +42,29 @@ def __radd__(self, other):
         return other
 
 
+@dataclass
+class ImportanceConfig:
+    """Used to provide parameters for importance sampling."""
+
+    importance_scores: Dict[Hashable, float] = None
+    tolerance: float = 10e-8
+    top_k: Dict[int, int] = None
+
+
+@dataclass
+class ConvergenceLog:
```

Comment:
Can the iterative process be exposed to the user, similar to an [optimizer](https://docs.pennylane.ai/en/stable/code/api/pennylane.GradientDescentOptimizer.html)?